From 25ca2ed56fd02d565a98ba43161af3d79bacda51 Mon Sep 17 00:00:00 2001
From: Igal Zeltser <izeltser@habana.ai>
Date: Wed, 06 Apr 2022 14:15:43 +0300
Subject: [PATCH] [SW-69581] gaudi3: relocate sram for onix

On Onix SRAM memory is carved out of HBM region and mapped to
different address than other setups.
In order to use the Sram from FW, the Sram extension registers
must be adjusted respectively.

Change-Id: I9fa05c169e175682bebc10adb5289cce874780e8
Signed-off-by: Igal Zeltser <izeltser@habana.ai>
---

diff --git a/arch/gaudi3/arc/onix.c b/arch/gaudi3/arc/onix.c
index 78abd17..ff3c659 100644
--- a/arch/gaudi3/arc/onix.c
+++ b/arch/gaudi3/arc/onix.c
@@ -4,6 +4,7 @@
  */
 
 #include <d0_psoc_global_conf_regs.h>
+#include <d0_parc_arc1_cfg_regs.h>
 
 #include <preboot.h>
 #include <uart.h>
@@ -36,3 +37,16 @@
 
 	return (factor + 1) & 0xFFFE;
 }
+
+/*
+ * SRAM on Onix is carved out of HBM region and mapped at address
+ * 0x0201_0002_0000_0000,
+ * change SRAM extension for ARC1 accordingly.
+ */
+#define SRAM_EXTENSION_MSB      0x2010002
+
+void sram_set_extension(void)
+{
+	regmap_writel(SRAM_EXTENSION_MSB, mmD0_PARC_ARC1_CFG_SRAM_MSB_ADDR);
+	regmap_writel(0x0, mmD0_PARC_ARC1_CFG_SRAM_LSB_ADDR);
+}
